packages/react-toast/src/hooks.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import {Toast} from '@shopify/polaris';
+
+export const ToastContext = React.createContext<ToastActions>({
+  show: noop,
+  hide: noop,
+});
+
+export interface ToastDescription {
+  content: string;
+  error?: boolean;
+  dismissible?: boolean;
+  duration?: number;
+  onDismiss?(): void;
+}
+
+export interface ToastActions {
+  show(options: ToastDescription): void;
+  hide(): void;
+}
+
+export interface Props {
+  children: any;
+}
+
+export function ToastProvider({children}: Props) {
```

Comment:
Nit: Can we split the context stuff into `/context`? Seems weird to me to have a file called `hooks` containing jsx.